artemis/modules/api_scanner.py

```diff
@@ -0,0 +1,162 @@
+import json
+import os
+import subprocess
+from typing import Any, Dict, Optional, Tuple
+
+from karton.core import Task
+from openapi_spec_validator import validate
+from openapi_spec_validator.readers import read_from_filename
+from pydantic import BaseModel
+
+from artemis import http_requests, load_risk_class
+from artemis.binds import Service, TaskStatus, TaskType
+from artemis.config import Config
+from artemis.module_base import ArtemisBase
+from artemis.modules.data.api_scanner_data import COMMON_SPEC_PATHS, VULN_DETAILS_MAP
+from artemis.task_utils import get_target_url
+
+
+class APIResult(BaseModel):
+    url: str
+    endpoint: str
+    data_leak: Optional[Dict[Any, Any]]
+    method: str
+    vulnerable: bool
+    vuln_details: Optional[str]
+    curl_command: Optional[str]
+    status_code: Optional[int]
+
+
+@load_risk_class.load_risk_class(load_risk_class.LoadRiskClass.MEDIUM)
+class APIScanner(ArtemisBase):
+    """
+    Scans API endpoints for vulnerabilities based on OpenAPI/Swagger specifications.
+    """
+
+    identity = "api_scanner"
+    filters = [
+        {"type": TaskType.SERVICE.value, "service": Service.HTTP.value},
+    ]
+
+    def discover_spec(self, base_url: str) -> Tuple[str, ...]:
+        """Try to discover OpenAPI/Swagger specification from common paths."""
+        for path in COMMON_SPEC_PATHS:
+            try_url = base_url.rstrip("/") + "/" + path
+            try:
+                response = http_requests.get(try_url)
+                if response.status_code == 200 and (
+                    "openapi" in response.text.lower() or "swagger" in response.text.lower()
+                ):
+                    temp_file = f"/tmp/api_spec_{os.urandom(8).hex()}"
```

Comment:
We need to remove the temp file so that we don't pollute the filesystem

Reply:
from what I see, it will be removed only if it's a real swagger file, it will be kept if it's something went wrong (e.g. validation failed). Or I'm missing somthing?